templates/master/00-master/gcp/files/opt-libexec-openshift-gcp-routes-sh.yaml

```
@@ -27,7 +27,7 @@ contents:
     declare -A vips
 
     curler() {
-      curl --silent -L -H "Metadata-Flavor: Google" "http://metadata.google.internal/computeMetadata/v1/instance/${1}"
+      BODY="$(curl --fail-with-body --silent -L -H "Metadata-Flavor: Google" "http://metadata.google.internal/computeMetadata/v1/instance/${1}")" && printf "%s" "${BODY}" || printf "%s" "${BODY}" >&2
```

```suggestion
      BODY="$(curl --fail-with-body --silent -L -H "Metadata-Flavor: Google" "http://metadata.google.internal/computeMetadata/v1/instance/${1}")" ||: 
      printf "%s" "${BODY}"  >&2
```
Is a bit cleaner since it clear that the error is completely ignored. I found the duplicate `prinft` confusing.

---

> I found the duplicate `prinft` confusing.

The success `printf` was writing to stdout (so the `curler` caller would have it available for consumption) while the failure `printf` was writing to stderr (so it would be available in unit logs, but would not be consumed by the `curler` caller).  Your alternative would write to stderr regardless, right?

---

> Your alternative would write to stderr regardless, right?

Correct. `|| :` effectively reads as "or nothing", effective meaning that `$BODY` is written regardless of success.

---

who should understand this code in 6 months? Please doc this.

---

The colon is a POSIX command [1].

```
4.1 Bourne Shell Builtins
The following shell builtin commands are inherited from the Bourne Shell. These commands are implemented as specified by the POSIX standard.

: (a colon)
    : [arguments]
Do nothing beyond expanding arguments and performing redirections. The return status is zero.
```

[1] https://www.gnu.org/software/bash/manual/html_node/Bourne-Shell-Builtins.html

---

yeah, but these options are not well known for everybody.
I  agree with Stefan and I think that we should add a comment explaining it
```suggestion
     # if the curl fails we write the body error to stderr only, to avoid failing the script
     # | : effectively reads as "or nothing" and prevents the failure from being bubbled 
     # and failing the job allowing the printf statement to execute
      BODY="$(curl --fail-with-body --silent -L -H "Metadata-Flavor: Google" "http://metadata.google.internal/computeMetadata/v1/instance/${1}")" && printf "%s" "${BODY}" || printf "%s" "${BODY}" >&2
```

---

rebased on master and added a longer comment based more on the commit message with a3817026 -> 4e263081.